### PR TITLE
New hasPrimaryKey and hasForeignKey implementations for SQLite

### LINKED ISSUE #1538: SQLite: hasPrimaryKey() fails variously
The `SQLiteAdapter::hasPrimaryKey()` method, (implemented largely by `SQLiteAdapter::getPrimaryKey()`) is only able to find very specifically-formatted primary keys and has no support for composite keys at all.

### LINKED ISSUE #1539: SQLite: hasForeignKey() fails variously
Similar to issue #1538, the `SQLiteAdapter::hasForeignKey` method fails in a varied number of ways:

- It can return `true` when asking about a composite key while only simple keys exist on the same columns
- It can return `true` when asking about a key while only a composite key containing a superset of the queried columns exists
- Identifiers are treated as case-sensitive when SQLite does not do so in the course of normal operation
- Any table definition which does not use uppercase `REFERENCES` will yield `false`
- It will answer `true` for a table does does not contain a key if a temporary table of the same name does
- It always returns `false` if the table name contains a schema
- It always returns `false` if the table name starts with the substring `sqlite` _not_ followed by an underscore
- It probably fails spectacularly if a trigger attached to a table includes a string literal with the substring `REFERENCES`, but I have not confirmed this

### PR DESCRIPTION
**Pull request by JKingweb**
Corrected implementations of `SQLiteAdapter::hasPrimaryKey()` and `SQLiteAdapter::hasForeignKey()`. They were changed in the following ways:

- Avoids haphazard SQL scraping
- Reports correctly on composite primary keys
- Reports correctly on composite foreign keys
- Reports correctly on subsets of composite keys
- Fully case-insensitive
- Fails noisily when a named constraint is queried

Fixes #1538 and  fixes #1539.

### CODE REVIEW

File: src/Phinx/Db/Adapter/SQLiteAdapter.php [hunk 1]
```diff
@@ -784,70 +804,67 @@ protected function getDropIndexByNameInstructions($tableName, $indexName)
...
+        $primaryKey = array_map('strtolower', $this->getPrimaryKey($tableName));
 
-        return empty($missingColumns);
+        if (array_diff($primaryKey, $columns)) {
+            return false;
+        } elseif (array_diff($columns, $primaryKey)) {
+            return false;
+        } else {
+            return true;
+        }
```
Comment (lorenzo) [inline src/Phinx/Db/Adapter/SQLiteAdapter.php]:
```php
if (array_diff($primaryKey, $columns)) {
    return false;
}
if (array_diff($columns, $primaryKey)) {
    return false;
}
return true;
```
please

File: src/Phinx/Db/Adapter/SQLiteAdapter.php [hunk 2]
```diff
@@ -784,70 +804,67 @@ protected function getDropIndexByNameInstructions($tableName, $indexName)
...
-        return !array_diff((array)$columns, $foreignKeys);
+        foreach ($foreignKeys as $key) {
+            $key = array_map('strtolower', $key);
+            if (array_diff($key, $columns)) {
+                continue;
+            } elseif (array_diff($columns, $key)) {
+                continue;
+            } else {
+                return true;
+            }
```
Comment (lorenzo) [inline src/Phinx/Db/Adapter/SQLiteAdapter.php]:
```php
if (array_diff($key, $columns) || array_diff($columns, $key)) {
    continue;
} 
return true;
```

please

File: src/Phinx/Db/Adapter/SQLiteAdapter.php (hunk 1 above)
Comment (JKingweb) [reply src/Phinx/Db/Adapter/SQLiteAdapter.php]: I removed the else, and also did an or comparison as in the other case.

**Comment by lorenzo** [comment]
Thanks agains for such a great job!

**Pull request merged**